dll/win32/msv1_0/msv1_0.c

```diff
@@ -1281,6 +1294,10 @@ LsaApLogonUserEx2(IN PLSA_CLIENT_REQUEST ClientRequest,
             LogonInfo->Password.MaximumLength = 0;
         }
 
+        /* decode password */
+        if (LogonPassHash > 0)
+            RtlRunDecodeUnicodeString(LogonPassHash, &LogonInfo->Password);
+
         Status = RtlValidateUnicodeString(0, &LogonInfo->Password);
         if (!NT_SUCCESS(Status))
```

Comment:
Dumb question, here and maybe after:
Does it matter that the "decoded" password remains in memory?

Reply:
Good point. RtlEraseUnicodeString will be a friend: https://github.com/reactos/reactos/pull/2357/commits/f3230f88c61a0fbc57b037afab0b6c80579914cf

Reply:
Much better, though not exactly complete:
There is 1(+) `return` codepaths which will miss that.
Now, the password is erased even if it was not an encoded one. That is a change, though I don't know if it matters.
For a password, I would feel safer with `RtlSecureZeroMemory` than `RtlEraseUnicodeString`.
NB:
I wonder if it could help to decode the password into a separate variable and only copy it back if/as needed?